source/text_handler.cpp

```diff
@@ -387,9 +387,14 @@ spv_result_t AssemblyContext::binaryEncodeFloatingPointLiteral(
     spv_instruction_t* pInst) {
   const auto bit_width = assumedBitWidth(type);
   switch (bit_width) {
-    case 16:
-      return diagnostic(SPV_ERROR_INTERNAL)
-             << "Unsupported yet: 16-bit float constants.";
+    case 16: {
+      spvutils::HexFloat<FloatProxy<spvutils::Float16>> hVal(0);
+      if (auto error = parseNumber(val, error_code, &hVal,
+                                   "Invalid 16-bit float literal: "))
+        return error;
+      return binaryEncodeU32(
+          static_cast<uint32_t>(hVal.value().getAsFloat().get_value()), pInst);
```

Comment:
Might be worth mentioning that getAsFloat will return the spvutils::Float16, and in turn the get_value() will return a uint16_t, so the encoding is correct as per SPIR-V spec, i.e. the bits are in the least significant 16 bits, and the most significant 16 bits are all 0.